src/language-css/printer-postcss.js

```diff
@@ -828,15 +828,16 @@ function printNodeSequence(path, options, print) {
 
     if (i !== node.nodes.length - 1) {
       if (
-        (node.nodes[i + 1].type === "css-comment" &&
+        node.nodes[i].type !== "front-matter" &&
+        ((node.nodes[i + 1].type === "css-comment" &&
           !hasNewline(
             options.originalText,
             options.locStart(node.nodes[i + 1]),
             { backwards: true }
           )) ||
-        (node.nodes[i + 1].type === "css-atrule" &&
-          node.nodes[i + 1].name === "else" &&
-          node.nodes[i].type !== "css-comment")
+          (node.nodes[i + 1].type === "css-atrule" &&
+            node.nodes[i + 1].name === "else" &&
+            node.nodes[i].type !== "css-comment"))
       ) {
         parts.push(" ");
       } else {
```

Comment:
There's a potential bug in:

https://github.com/prettier/prettier/blob/56af3d621d6890819d65d4c3604425f187e7643a/src/language-css/printer-postcss.js#L842-L849

where `isNextLineEmpty` _could_ be `true` for `front-matter`, but it's currently always `false` since there's no loc info in `front-matter`:

https://github.com/prettier/prettier/blob/56af3d621d6890819d65d4c3604425f187e7643a/src/language-css/parser-postcss.js#L500-L503

Two wrongs make the thing right, I think we should fix it somehow? (maybe in another PR)

Reply:
@ikatyang can you give example where potential we have bug?

Reply:
I meant, if we fix the bug that there's no loc info in `front-matter`, then it'll become a bug.

```diff
 result.nodes.unshift({
   type: "front-matter",
   value: frontMatter,
+  source: {
+    start: { line: 1, column: 1 },
+    end: {
+      line: frontMatter.split("\n").length,
+      column: frontMatter.split("\n").pop().length
+    },
+    input: { css: frontMatter }
+  }
 });
```

```diff

 ---
 ---
 
 /** @prettier */
 
 .class {    display: none;    }
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 ---
 ---
 
+
 /** @prettier */
 .class {
   display: none;
 }
```